### PR TITLE
Added test case to check all jobs succeded

### PR DESCRIPTION
**Pull request by AverageMarcus**
### What this PR does

Fixes: https://github.com/giantswarm/giantswarm/issues/31140

Adds a new basic check to ensure all Jobs in the WC have succeded.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

**Review by fiunchinho — APPROVED** [review]

File: internal/common/basic.go [hunk 1]
```diff
@@ -239,6 +247,35 @@ func CheckAllStatefulSetsReady(wcClient *client.Client) func() error {
 	}
 }
 
+func CheckAllJobsSucceeded(wcClient *client.Client) func() error {
```
[the remark this answers is not in the document]
Comment (AverageMarcus) [reply internal/common/basic.go:250]: Yeah, that makes sense. I hadn't realised it didn't

[the remark this answers is not in the document]
Comment (AverageMarcus) [reply internal/common/basic.go]: I decided to do it as part of this PR.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Added a test case to check the status of all Jobs in the WC
```
[the remark this answers is not in the document]
Comment (AverageMarcus) [reply CHANGELOG.md:12]: Good point! Updated :) (Not going to bother running the tests against the new commit unless they fail)

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**